tests/test_javascript_server.py

```diff
@@ -67,3 +72,61 @@ def test_generate_ping_factory_method():
     expected_result = "createAccountsEventsEvent"
     result = javascript_server.generate_ping_factory_method(ping)
     assert result == expected_result
+
+
+def run_logger(code_dir, import_file, factory, code):
+    """
+    Run the JavaScript logger with a mocked logger
+    that just prints the ping payload to STDOUT.
+    """
+
+    shutil.copy(ROOT / "test-js" / "package.json", code_dir)
+    subprocess.check_call(["npm", "install"], cwd=code_dir)
+
+    tmpl_code = ""
+    with open(ROOT / "test-js" / "test.js.tmpl", "r") as fp:
+        tmpl_code = fp.read()
+
+    tmpl_code = (
+        tmpl_code.replace("/* IMPORT */", import_file)
+        .replace("/* FACTORY */", factory)
+        .replace("/* CODE */", code)
+    )
+
+    with open(code_dir / "test.js", "w") as fp:
+        fp.write(tmpl_code)
+
+    return subprocess.check_output(["node", "test.js"], cwd=code_dir).decode("utf-8")
+
+
+@pytest.mark.web_dependency
+def test_run_logging(tmpdir):
+    tmpdir = Path(str(tmpdir))
+
+    translate.translate(
+        [
+            ROOT / "data" / "fxa-server-pings.yaml",
+            ROOT / "data" / "fxa-server-metrics.yaml",
+        ],
+        "javascript_server",
+        tmpdir,
+    )
+
+    factory = "createAccountsEventsEvent"
+    code = """
+events.record({ user_agent: "glean-test/1.0", event_name: "testing" });
```

Comment:
Oh just realized the variable name here is still hardcoded, so we might want to shuffle around what code needs to be provided a bit. Or if we really just stick to one test here we don't bother with templates and just include the full file.